src/techonomy/django/django_tableselect/static/django_tableselect/tableselect.js

```diff
@@ -0,0 +1,123 @@
+// TODO: make this code compatible with dinosaurs (old browsers)
+addEventListener("DOMContentLoaded", () => {
+  const tableSelector = "[data-tableselect]";
+  const bulkCheckboxSelector = "input[type=checkbox][data-bulk-select]";
+  const ariaNarratorSelector = "[data-tableselect-narrator]";
+  const selectCheckboxSelector = "input[type=checkbox][name]";
+
+  const allSelectedTextAttribute = "data-trans-all-selected";
+  const selectAllSelectedTextAttribute = "data-trans-select-all";
+  const deselectAllSelectedTextAttribute = "data-trans-deselect-all";
+  const numSelectedTextAttribute = "data-trans-num-selected";
+
+  const allChecked = (nodes) => Array.from(nodes).every((node) => node.checked);
+  const partiallyChecked = (nodes) =>
+    Array.from(nodes).some((node) => node.checked);
+
+  const tables = document.querySelectorAll(tableSelector);
+
+  tables.forEach((table) => {
+    const bulkCheckbox = table.querySelector(bulkCheckboxSelector);
+    const selectCheckBoxes = table.querySelectorAll(selectCheckboxSelector);
+    const ariaNarrator = table.querySelector(ariaNarratorSelector);
+
+    let initialized = false;
+
+    // No bulk checkbox = bulk select not enabled.
+    // No point in initializing further.
+    if (!bulkCheckbox) return;
+
+    // Get text placeholders from data attributes
+    const selectAllSelectedPlaceholder = bulkCheckbox.getAttribute(
+      selectAllSelectedTextAttribute
+    );
+    const deselectAllSelectedPlaceholder = bulkCheckbox.getAttribute(
+      deselectAllSelectedTextAttribute
+    );
+    const allSelectedPlaceholder = bulkCheckbox.getAttribute(
+      allSelectedTextAttribute
+    );
+    const numSelectedPlaceholder = bulkCheckbox.getAttribute(
+      numSelectedTextAttribute
+    );
+
+    // Responsible for updating the checked state and accessible label of the checkbox
+    const updateBulkCheckboxState = () => {
+      let checked = partiallyChecked(selectCheckBoxes);
+
+      bulkCheckbox.indeterminate =
+        !allChecked(selectCheckBoxes) && partiallyChecked(selectCheckBoxes);
+      bulkCheckbox.checked = checked;
+
+      const totalRows = selectCheckBoxes.length;
+
+      if (checked) {
+        bulkCheckbox.ariaLabel = deselectAllSelectedPlaceholder.replace(
+          "%num%",
+          totalRows
+        );
+      } else {
+        bulkCheckbox.ariaLabel = selectAllSelectedPlaceholder.replace(
+          "%num%",
+          totalRows
+        );
+      }
+
+      // We only want to narrate screenreaders announcements when the page changes, not upon initialization
+      if (initialized) {
+        updateNarrator();
```

Comment:
Hierbij zou ik eventueel korter schrijven in een lijn of korte variant

initialized && updateNarator();

Reply:
Ik vind dit zelf minder leesbaar